lib/iris/_concatenate.py

```diff
@@ -926,6 +930,12 @@ def register(
                         or candidate_axis not in coord_b.dims
                     ):
                         if not coord_a == coord_b:
+                            mismatch_error_msg = (
+                                "Auxiliary coordinates differ for phenomenon"
```

Comment:
I'm looking at the `match()` method (code below). The word `differ` is used there too, meaning there's no decent distinction between those existing errors (where a coord is absent, or the metadata is different), and these new errors.

**I prefer the word `unequal` here**, because we are actually using equality whereas the existing code is doing more basic comparisons. That should give us the distinction we need.

https://github.com/SciTools/iris/blob/e92e36ac0e845318635a507b7e58914c5633d03c/lib/iris/_concatenate.py#L554-L562